src/core/data.jl

```diff
@@ -1347,10 +1354,11 @@ end
 
 "checks that all branches connect two distinct buses"
 function check_branch_loops(data::Dict{String,<:Any})
-    if _IM.ismultinetwork(data)
-        Memento.error(_LOGGER, "check_branch_loops does not yet support multinetwork data")
-    end
+    _IM.modify_data_with_function!(data, _pm_it_name, _check_branch_loops)
```

Comment:
As a general remark looking at the functions in this file, some have a simple workaround to support `_IM.ismultiinfrastructure(data_all) ? data_all["it"][_pm_it_name] : data_all`, others are generalized to support the multi-network case, for example `_IM.modify_data_with_function!(data, _pm_it_name, _check_branch_loops)`.

For these functions it is not always clear what the semantics should be in the multi-network case, so I would be inclined to not generalize the behavior and instead replicate the current functionality in each situation, using the new IM helpers when possible to replicate without generalizing the scope of the feature.

Reply:
Let's discuss more on Thursday. To the best of my recollection, I only used `_IM.modify_data_with_function!` when it was possible to generalize the function.